graylog2-server/src/main/java/org/graylog/scheduler/JobSchedulerService.java

```diff
@@ -69,14 +70,21 @@ public JobSchedulerService(JobExecutionEngine.Factory engineFactory,
     @Override
     protected void startUp() throws Exception {
         schedulerEventBus.register(this);
+        this.executionThread = Thread.currentThread();
     }
 
     @Override
     protected void run() throws Exception {
         // Safety measure to make sure everything is started before we start job scheduling.
         LOG.debug("Waiting for server to enter RUNNING status before starting the scheduler loop");
-        serverStatus.awaitRunning(() -> LOG.debug("Server entered RUNNING state, starting scheduler loop"));
-
+        try {
+            serverStatus.awaitRunning();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Hmm,  why do we need this?
The thread is already interrupted. Or am I missing something?

Reply:
Nevermind, found the answer myself. Weird semantics :/

Reply:
Are you referring to setting the interrupt flag with `Thread.currentThread().interrupt();`? In this particular case we don't strictly need it but afaik it's good practice to set the flag when the thread was interrupted to let the caller know about it.